plugins/gcp/fix_plugin_gcp/resources/base.py

```diff
@@ -173,25 +173,33 @@ def _standard_edges(self, node: GcpResourceType, source: Optional[Json] = None)
             return True
 
         if source is not None:
+            if "zone" in source:
+                zone_name = source["zone"].rsplit("/", 1)[-1]
+                if zone := self.zone_by_name.get(zone_name):
+                    node._zone = zone
+                    node._region = self.region_by_zone_name[zone_name]
+                    self.add_edge(node, node=zone, reverse=True)
+                    return True
+            if "region" in source:
+                region_name = source["region"].rsplit("/", 1)[-1]
+                if region := self.region_by_name.get(region_name):
+                    node._region = region
+                    self.add_edge(node, node=region, reverse=True)
+                    return True
+
             if InternalZoneProp in source:
                 if zone := self.zone_by_name.get(source[InternalZoneProp]):
                     node._zone = zone
                     node._region = self.region_by_zone_name[source[InternalZoneProp]]
                     self.add_edge(node, node=zone, reverse=True)
                     return True
-                else:
```

Comment:
why remove? a zone prop was defined, but no zone is found.

Reply:
If we return False, then we will not add the node to the graph:
```python
    def add_node(self, node: GcpResourceType, source: Optional[Json] = None) -> Optional[GcpResourceType]:
       ...
        if self._standard_edges(node, source):
            with self.graph_nodes_access:
                self.graph.add_node(node, source=source or {})
            return node
        return None
```
Also we have the another checking if the graph_builder has region property:
```python
        # Fallback to GraphBuilder region, i.e. regional collection
        if self.region is not None:
            node._region = self.region
            self.add_edge(node, node=self.region, reverse=True)
            return True
 ```
In case if region still not found, we just add the `global` region:
```python
        # Fallback to global region
        node._region = self.fallback_global_region
        self.add_edge(node, node=self.fallback_global_region, reverse=True)
        return True
```
This way the node will always be added to the graph

Reply:
If a zone prop is defined but the related zone is not found, the builder will attach it to its region.
The error will silently be dropped. 
We should at least log a warning in such a case.